File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/DTOS/CartDTOS/CartDTO.java
package app.aurafitbackend.DTOS.CartDTOS;

import app.aurafitbackend.Enums.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class CartDTO {

    private Long id;
    private List<CartItemDTO> items;
    private BigDecimal shippingCost;
    private BigDecimal subTotal;
    private BigDecimal totalCartPrice;
    private Status status;
    private Long userId;
    private String cartToken;


}

File: aurafit-frontend/src/Models/DTOS/CartDTO.ts
import { CartItemDTO } from "./CartItemDTO";
import {Status} from "../Enums/Status.ts";


export class CartDTO {
    id: number;
    items: CartItemDTO[];
    shippingCost: number;
    totalCartPrice: number;
    subTotal: number;
    status: Status;
    userId: number | null;
    cartToken: string | null;

    constructor(id: number, items: CartItemDTO[], shippingCost: number, subTotal: number ,totalCartPrice: number, status: Status, userId: number, cartToken: string | null) {
        this.id = id;
        this.items = items;
        this.shippingCost = shippingCost;
        this.subTotal = subTotal;
        this.totalCartPrice = totalCartPrice;
        this.status = status;
        this.userId = userId;
        this.cartToken = cartToken;

    }
}